src/core/event.cpp

```diff
@@ -364,7 +364,9 @@ void on_cell_structure_change() {
 #endif /* ifdef DIPOLES */
 
 #ifdef LB
-  lb_lbfluid_init();
+  if (lattice_switch == ActiveLB::CPU) {
```

Comment:
Shouldn't this call some interface function, instead of checking the lattice switch? Also how sure are you that no action should be taken for the GPU code in this case?